joint_trajectory_controller/include/joint_trajectory_controller/init_joint_trajectory.h

```diff
@@ -314,6 +314,16 @@ Trajectory initJointTrajectory(const trajectory_msgs::JointTrajectory&       msg
                       "s in the past.");
       return Trajectory();
     }
+    else if ( // If the first point is at time zero and no start time is set in the header, skip it silently
+              msg.points.begin()->time_from_start == ros::Duration(0.) &&
+              msg.header.stamp == ros::Time(0) &&
+              std::distance(msg.points.begin(), msg_it) == 1
```

Comment:
Not sure about this clause.
This would be valid only for the second point in the trajectory(?).

https://github.com/ros-controls/ros_controllers/blob/7d2f98db49552cab0af753421955071c3cbae8e4/joint_trajectory_controller/include/joint_trajectory_controller/init_joint_trajectory.h#L307 seems to skip the first point.

Reply:
I've copied that from two lines below: Distance of 1 means dropping one point (starting at the second one).

Reply:
Im missed the fact that you were checking `msg.points.begin()`.